### ISSUE TITLE
NVDA Alpha 29119 disables or removes ColumnsReview, and modifying manifest.ini doesn't help either

### ISSUE THREAD

**Issue by amirsol81** [ABuffEr/columnsReview#35]
Hi Alberto,

NVDA Alpha 29119 is an API-breaking release, updating the API requirement to 2024.1. However, even after altering ColumnsReview's manifest.ini, it can't work with this alpha - unlike other add-ons. Moreover, I no longer have ColumnsReview in my list of add-ons in the Add-on Store with alpha 29119 - it's disappeared! As such, can ColumnsReview be altered to work with this alpha - like other add-ons? As far as I know, so far NVDA hasn't altered ColumnsReviews requirements to generate a breakage like this.
Thanks.

**Comment by ABuffEr** [comment]
Hi @amirsol81,
thanks for reporting. I don't know exactly, but probably Alpha versions are integrating breaking changes that will merge in 2024.1. I read something around... there is also [an issue](https://github.com/nvaccess/nvda/issues/15401).
Give me a couple of days to fix.

**Comment by amirsol81** [comment]
Thanks @ABuffEr for your prompt reply - do take your time.
So far the only add-on breaking issue is that the IAccessibleHandler.SecureDesktopNVDAObject has been removed, but I'm not sure if ColumnsReview uses that. In the case of ColumnsReview, it's completely disappeared from my list of add-ons in the Add-on Store as if it were never installed.
Thanks

**Comment by amirsol81** [comment]
@ABuffEr Greetings.
I've got yet another interesting finding regarding this issue.
The simple act of modifying ColumnsReview's manifest.ini file and setting its date to 2024.1 forces the add-on to disappear from the list of add-ons regardless of whatever NVDA version being in use. So I tried this on another machine which hasn't yet been updated to Alpha 29119, and as soon as I altered the ini file to make the date change to 2024.1, restarting NVDA wiped ColumnsReview from the list of add-ons. Quite interestingly, RSSOwlnix isn't affected. Just thought it might help resolve the issue faster as it apparently isn't related to NVDA's recent API-breaking changes.

**Comment by ABuffEr** [comment]
Thanks for update. Hmm, I think it's something related to add-on store (where RSSOwlnix Support is not published). So it seems first a NVDA issue.
For curiosity, do you see ColumnsReview in available add-ons to install from add-on store? What's happen if you install it from there? I say it watching at the comment in the issue linked above...

**Comment by amirsol81** [comment]
Thanks, @ABuffEr for the explanation.
If I remove ColumnsReview first and then attempt to install it from the Add-on Store - ignoring the incompatibility warning, the add-on won't be installed either. However, after that, each time I open the Add-on Store and close it, it asks me to restart NVDA to finish the installation process - a process which won't be completed. Interestingly, I can't find the ColumnsReview folder in my Add-ons folder either - I just see a columnsReview.json file there.
It's worth mentioning that I've installed other add-ons both from inside and outside the Store, but modifying their manifest.ini has resolved the issue. ColumnsReview seems to be the only exception.
Thanks again.

**Comment by amirsol81** [comment]
@ABuffEr Greetings.
Given the release of a few more NVDA alphas (29163 now), I'm wondering if any progress has been made regarding this. I still can't use ColumnsReview with newer alphas on Win 10 and 11, and installing from the Add-on Store doesn't help either. As it's installed from the Store, it remains in a constant state of "restart is required," but restarting NVDA doesn't resolve the issue.
Thanks.

**Comment by ABuffEr** [comment]
Hi @amirsol81,
I installed NVDA alpha-29167, receiving the warning about incompatible add-ons. Then I chose ColumnsReview from installed add-ons, and enabled it ignoring compatibility, from context menu, that I suspect it's the correct way now. All worked.
Unfortunately, I'm afraid you fell in a corrupted state of add-on store, as it seems reported in some issues (there were a lot in last days about it, and sadly some ones closed as not planned/won't fix). So I have no explanation nor practical solution...
Have you tried to restore:
lastTestedNVDAVersion = 2023.2
in manifest, and then to see if add-on appears again?
Otherwise you can try to manually delete columnsReview folder under %appdata%\nvda\addons, and then install it again from the add-on store, hopefully...

**Comment by amirsol81** [comment]
@ABuffEr Thanks for the explanation.
I've tried everything you can imagine, but ColumnsReview doesn't work - unfortunately. I even downgraded NVDA to an older alpha - older than 29119, and then re-installed ColumnsReview from the Add-on Store. It worked as expected, but once I upgraded NVDA to Alpha 29167, ColumnsReview disappeared from my list of installed/incompatible add-ons. It's nowhere to be seen in the Store.
I even tried hand-deleting its folder, installing manually, installing from the Add-on store, but nothing helps. Interesting - and saddeningly, only ColumnsReview is affected here and other add-ons work.

**Comment by amirsol81** [comment]
@ABuffEr Finally I did it!
After almost everything failed, I re-downloaded the latest development release of ColumnsReview, altered its format to ZIP, unzipped it, altered Manifest.ini to reflect the 2024.1 requirement, re-zipped everything, altered the format of the ZIP file to .nvda-addon, and re-installed it. This time around, NVDA restarted and now I have ColumnsReview functioning properly with Alpha 29167. Moreover, opening/closing the Add-on Store no longer forces NVDA to display the "restart needed" message. I'm glad I restored the functionality!
So @ABuffEr I'm closing this issue. Thanks for your assistance.
